Tests/xcodeprojMeasurementsTests/MeasurementsTests.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import PathKit
+import SwiftShell
+@testable import xcodeproj
+import XCTest
+
+final class MeasurementsTests: XCTestCase {
+    
+    let tempDirectory = Path.temporary
+    var projectPath: Path?
+    var sourceRoot: Path?
+    
+    override func setUp() {
+        super.setUp()
+        try? tempDirectory.delete()
+        self.projectPath = tempDirectory + Path("Project")
+        self.sourceRoot = tempDirectory + Path("Sources")
+        let projectPath = self.projectPath!
+        let sourceRoot = self.sourceRoot!
+        let xcodeproj = try! generateXcodeProj(path: projectPath, sourceRoot: sourceRoot)
+        validate(xcodeproj: xcodeproj)
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        try? tempDirectory.delete()
+    }
+    
+    func test_measureProjectRead() {
+        let duration = try! measureDuration {
+            _ = try XcodeProj(path: self.projectPath!)
+        }
+        XCTAssertLessThan(duration, 3)
+    }
+    
+    func test_measureProjectWrite() {
+        let xcodeproj = try! XcodeProj(path: self.projectPath!)
+        let duration = try! measureDuration {
+            try xcodeproj.write(path: self.projectPath!)
+        }
+        XCTAssertLessThan(duration, 5)
+    }
+    
+    func test_measureObtainFilesFullPath() {
+        let xcodeproj = try! XcodeProj(path: self.projectPath!)
+        var pathList = [Path]()
+        var count = 0
+        let duration = try! measureDuration {
+            for project in xcodeproj.pbxproj.projects {
+                for target in project.targets {
+                    for buildPhase in target.buildPhases {
+                        let files = buildPhase.files
+                        for file in files {
```

Comment:
@pepibumur Here is the problem. Maybe you can tell me how to solve it.
Array of files is empty. I found out that when I save a project file, an object is released for all the build files. ( `throw PBXObjectError.objectsReleased` ). But I did not understand the reason. Most likely some kind of error when generating.